internal/auth/middleware.go

```diff
@@ -50,21 +84,28 @@ func CreatePermissionsMiddleware(log logrus.FieldLogger) func(http.Handler) http
 			} else {
```

Comment:
Setting `actionGet` irrespective of the `r.Method` means using other methods on `/api/version` would pass permission check, too. Handlers for those don't exist, of course, but still feels a bit off.

If you want to fix, a quick way could be to look up the defaultAction first, then pass the action (instead of `r.Method`) into extractResourceAndAction() that would modify the action if needed.

Reply:
Thanks for the suggestion. I've updated the logic to use the `defaultAction`

Reply:
Also note that I am not calling `extractResourceAndAction` afterwards which would return `NilAction` as it is trying to match the `method` passed to `defaultActions` [see](https://github.com/flightctl/flightctl/blob/main/internal/auth/permissions.go#L99-L102)